js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -948,7 +937,25 @@ const EditingController = modules.ViewController.inherit((function() {
         _beforeUpdateItems: function() { },
 
         _getVisibleEditRowIndex: function() {
-            return this._editRowIndex >= 0 ? this._editRowIndex - this._dataController.getRowIndexOffset() : -1;
+            const dataController = this._dataController;
+            const columns = this._columnsController.getColumns();
+
+            let editColumn;
+            columns.some(column => {
+                if(column.visibleIndex === this._editColumnIndex) {
+                    editColumn = column;
+                    return true;
+                }
+            });
+            const isColumnHidden = editColumn?.visibleWidth === 'adaptiveHidden';
+
+            const editRowIndexCorrection = isColumnHidden ? 1 : 0;
+
+            return dataController.getRowIndexByKey(this._editRowKey) + editRowIndexCorrection;
```

Comment:
if editRowKey is not exist on current page and edit column is hidden the result will be incorrect

Reply:
if this._editRowKey is defined `dataController.getRowIndexByKey(this._editRowKey)` can be equal -1 if key is on another page. In this case -1 + 1 = 0.